### PR TITLE
Support selecting between version in the docs

### LINKED ISSUE #1: Support selecting between versions in the docs
The new documentation is based on Docusaurus. Since hoprd has many releases we want to be able to present the different versions of the documentation through the same page. A user should be able to select which version is presented through a drop-down field or an alternative mechanism.

# User Requirements

- be able to access `master` branch documentation
- be able to access `release/*` branch documentation for each release

### PR DESCRIPTION
**Pull request by KedziaPawel**
**Solution overview:**
In order to have information about the releases, I've used the GitHub API.
Unfortunately, Docosaurus doesn't support custom react components in the navbar. That's why I've used the react portal to add versions to the dropdown list.

Flow proposition:
1. Developer creates branch name "release/X.X.X".
2. Developer does local changes and pushes the branch.
3. The pipeline triggers (Github Actions), builds, deploys the documentation to the "https://docs.hoprnet.org/X.X.X" and tags release as X.X.X (deployment and tagging could be also trigged manually, it depends on the team needs).
4. User refreshes the docs page, the information about the new release is fetched from GitHub and added to versions list.

Note: 
I have assumpted that master would be deployed to the "https://docs.hoprnet.org/latest".

The biggest advantage of this is solution is that it is automatic. There is no need to make any manual actions, for example, rebuild and redeploy the previous releases so they would have the latest versions in the dropdown.

**Useful links related to PR:**
https://docusaurus.io/docs/api/themes/configuration#navbar-items
https://reactjs.org/docs/portals.html
https://docs.github.com/en/rest/reference/repos#releases

Closes #1

### CODE REVIEW

File: src/api/github/consts.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export const BASE_URL = "https://api.github.com/repos/hoprnet/hopr-docs";
+
+export const ENDPOINTS = {
```
Comment (tolbrino) [inline src/api/github/consts.js:3]: Please add a link to the respective API docs.

File: src/portals/DocsVersion.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import React, { useEffect, useState } from "react";
```
Comment (tolbrino) [inline src/portals/DocsVersion.js:1]: Please document the component so the next person understands its use.

**Review by tolbrino — CHANGES_REQUESTED** [review]
Thanks! The approach is sound. Could you add the requested documentation?

File: src/api/github/consts.js (hunk 1 above)
Comment (KedziaPawel) [reply src/api/github/consts.js:4]: done

File: src/portals/DocsVersion.js (hunk 2 above)
Comment (KedziaPawel) [reply src/portals/DocsVersion.js:6]: done

**Comment by QYuQianchen** [comment]
Hey @KedziaPawel Have you "submit work" on Gitcoin?

**Comment by tolbrino** [comment]
Approach was clear and used as input for the final version. Closing this PR.

**Pull request closed without merging**